File: practice/mathematics/number_theory/dancing_in_pairs/dancing_in_pairs.hs
module Main where

import Control.Monad (forM_)

isqrt :: Integer -> Integer
isqrt = floor . sqrt . fromIntegral

solve :: Integer -> String
solve n
    | a `mod` 2 == 0 = "even"
    | otherwise      = "odd"
    where
        a' = isqrt n
        a  = if a' * a' > n then a' - 1 else a'

main :: IO ()
main = do
    t <- readLn :: IO Int
    forM_ [1..t] $ \_ -> do
        n <- readLn :: IO Integer
        putStrLn $ solve n

File: practice/mathematics/number_theory/dancing_in_pairs/dancing_in_pairs.cpp
#include <cmath>
#include <iostream>

using namespace std;

int main()
{
    int t;
    unsigned long long n, a;
    
    cin >> t;
    while (t--)
    {
        cin >> n;
        a = static_cast<unsigned long long>(sqrt(n));
        if (a * a > n)
        {
            a--;
        }
        
        if (a % 2 == 0)
        {
            cout << "even" << endl;
        }
        else
        {
            cout << "odd" << endl;
        }
    }
    
    return 0;
}

